src/buildercore/bootstrap.py

```diff
@@ -415,8 +413,9 @@ def download_master_builder_key(stackname):
     master_stack = core.find_master(region)
     private_key = "/root/.ssh/id_rsa"
     with stack_conn(master_stack):
-        with show('exceptions'): # I actually get better exceptions with this disabled
-            return fab_get(private_key, use_sudo=True, return_stream=True, label="master builder key %s:%s" % (master_stack, private_key))
+        # with fabric.api.show('exceptions'): # I actually get better exceptions with this disabled
```

Comment:
test these exceptions

Reply:
locally the exceptions are identical:

```
exception while executing task 'update': fooooooooooo

Traceback (most recent call last):
  File "src/taskrunner.py", line 214, in exec_task
    return_map['result'] = task_map['fn'](*task_args, **task_kwargs)
  File "/home/luke/dev/python/experimental/builder/src/decorators.py", line 80, in call
    return func(stackname, *args, **kwargs)
  File "/home/luke/dev/python/experimental/builder/src/decorators.py", line 19, in wrap
    result = fn(*args, **kw)
  File "/home/luke/dev/python/experimental/builder/src/cfn.py", line 57, in update
    return bootstrap.update_stack(stackname, service_list=['ec2'], concurrency=concurrency)
  File "/home/luke/dev/python/experimental/builder/src/buildercore/bootstrap.py", line 397, in update_stack
    fn(stackname, context, **actual_arguments)
  File "/home/luke/dev/python/experimental/builder/src/buildercore/context_handler.py", line 84, in decorated_with_only_if
    return fn(stackname, context, **kwargs)
  File "/home/luke/dev/python/experimental/builder/src/buildercore/decorators.py", line 18, in _wrapper
    return func(stackname, *args, **kwargs)
  File "/home/luke/dev/python/experimental/builder/src/buildercore/bootstrap.py", line 482, in update_ec2_stack
    master_builder_key = download_master_builder_key(stackname)
  File "/home/luke/dev/python/experimental/builder/src/buildercore/bootstrap.py", line 419, in download_master_builder_key
    raise ValueError("fooooooooooo")
ValueError: fooooooooooo
```

I'll see if I can trigger a remote exception or something

Reply:
I think that comment may be right:

![Screenshot at 2019-11-04 17-22-16](https://user-images.githubusercontent.com/2142748/68103650-ae06cf00-ff27-11e9-93c4-6a7b9913c126.png)

Left side is with `with fabric.api.show('exceptions'):` commented out. 

Right side is with `with fabric.api.show('exceptions'):` enabled.

Reply:
you can use this to recreate if curious. 

```patch
diff --git a/src/cfn.py b/src/cfn.py
index c1a6e174..93506c9c 100644
--- a/src/cfn.py
+++ b/src/cfn.py
@@ -340,15 +340,17 @@ def cmd(stackname, command=None, username=DEPLOY_USER, clean_output=False, concu
         custom_settings['output_prefix'] = False
 
     try:
+        import fabric.api
         with settings(**custom_settings):
-            return stack_all_ec2_nodes(
-                stackname,
-                (run, {'command': command}),
-                username=username,
-                abort_on_prompts=True,
-                concurrency=concurrency_for(stackname, concurrency),
-                node=int(node) if node else None
-            )
+            with fabric.api.show('exceptions'):
+                return stack_all_ec2_nodes(
+                    stackname,
+                    (run, {'command': command}),
+                    username=username,
+                    abort_on_prompts=True,
+                    concurrency=concurrency_for(stackname, concurrency),
+                    node=int(node) if node else None
+                )
     except FabricException as e:
         LOG.error(e)
         exit(2)
```